pkg/v7/configmap/desired.go

```diff
@@ -21,44 +21,48 @@ func (s *Service) GetDesiredState(ctx context.Context, clusterConfig ClusterConf
 	configMapSpecs := newConfigMapSpecs(providerChartSpecs)
 
 	for _, spec := range configMapSpecs {
-		values := []byte{}
-
-		switch spec.App {
-		case "cert-exporter":
-			values, err = exporterValues(configMapValues)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
-		case "coredns":
-			values, err = coreDNSValues(configMapValues)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
-		case "net-exporter":
-			values, err = exporterValues(configMapValues)
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
-		case "nginx-ingress-controller":
-			releaseExists, err := s.checkHelmReleaseExists(ctx, spec.ReleaseName, clusterConfig)
-			if err != nil {
-				return nil, microerror.Mask(err)
+		spec.Labels = newConfigMapLabels(spec, configMapValues, s.projectName)
+
+		// Values are only set for app configmaps.
+		if spec.Type == label.ConfigMapTypeApp {
+			values := []byte{}
+
+			switch spec.App {
+			case "cert-exporter":
+				values, err = exporterValues(configMapValues)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+			case "coredns":
+				values, err = coreDNSValues(configMapValues)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+			case "net-exporter":
+				values, err = exporterValues(configMapValues)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+			case "nginx-ingress-controller":
+				releaseExists, err := s.checkHelmReleaseExists(ctx, spec.ReleaseName, clusterConfig)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+
+				values, err = ingressControllerValues(configMapValues, releaseExists)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+			default:
```

Comment:
Yes its for AWS China support. We use the Aliyun registry because pulling from Quay is too slow.

The values configmap is optional. You can just leave it out of the `ConfigMapSpec` in the key package. We do that for the `external-dns` chart we create for Azure.